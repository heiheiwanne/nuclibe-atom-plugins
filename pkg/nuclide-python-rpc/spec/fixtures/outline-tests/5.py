#!/usr/bin/env python3
from typing import TypeVar, Generic


T = TypeVar('T')


class Node(Generic[T]):
    def __init__(self, label: T = None) -> None:
        pass


def fun(arg1: float, arg2: float) -> float:
    pass


def fun2(a, *, b):
    pass
