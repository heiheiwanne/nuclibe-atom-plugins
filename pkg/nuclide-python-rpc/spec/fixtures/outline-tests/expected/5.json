[
  {
    "end": {
      "column": 16,
      "line": 5
    },
    "kind": "statement",
    "name": "T",
    "start": {
      "column": 0,
      "line": 5
    }
  },
  {
    "children": [
      {
        "children": [],
        "end": {
          "column": 0,
          "line": 13
        },
        "kind": "function",
        "name": "__init__",
        "params": [
          "self",
          "label"
        ],
        "start": {
          "column": 4,
          "line": 9
        }
      }
    ],
    "end": {
      "column": 0,
      "line": 13
    },
    "kind": "class",
    "name": "Node",
    "params": [
      "label"
    ],
    "start": {
      "column": 0,
      "line": 8
    }
  },
  {
    "children": [],
    "end": {
      "column": 0,
      "line": 17
    },
    "kind": "function",
    "name": "fun",
    "params": [
      "arg1",
      "arg2"
    ],
    "start": {
      "column": 0,
      "line": 13
    }
  },
  {
    "children": [],
    "end": {
      "column": 0,
      "line": 19
    },
    "kind": "function",
    "name": "fun2",
    "params": [
      "a",
      "*",
      "b"
    ],
    "start": {
      "column": 0,
      "line": 17
    }
  }
]
